safe-logging/src/main/java/com/palantir/logsafe/SafeArg.java

```diff
@@ -25,7 +26,7 @@ private SafeArg(String name, @Nullable T value) {
         super(name, value);
     }
 
-    public static <T> SafeArg<T> of(String name, @Nullable T value) {
+    public static <T> SafeArg<T> of(@CompileTimeConstant String name, @Nullable T value) {
```

Comment:
I think this is pretty neat as the requirement is now visible in code not just in compilation failures. Realistically, all services internally already have error_prone_annotations on the classpath, and I don't think we've seen any problems with error_prone_annotations shipping breaks, so +1 from me.